core/metacling/src/TCling.cxx

```diff
@@ -6149,116 +6155,134 @@ static bool LookupNormalSymbols(llvm::object::ObjectFile *RealSoFile, const std:
    return false;
 }
 
-static void* LazyFunctionCreatorAutoloadForModule(const std::string& mangled_name,
-                                                  cling::Interpreter *fInterpreter) {
-   using namespace llvm::object;
+///\returns true on success
+static bool LookupDynamicSymbols(const std::string &library_filename, std::string mangled_name)
+{
+   auto ObjF = llvm::object::ObjectFile::createObjectFile(library_filename);
+   if (!ObjF)
+      return false;
+
+   llvm::object::ObjectFile *BinObjFile = ObjF.get().getBinary();
+
+   for (const auto &section : BinObjFile->sections()) {
+      llvm::StringRef sectionName;
+      section.getName(sectionName);
+
+      // .dynstr contains string of dynamic symbols
+      if (sectionName == ".dynstr") {
+         llvm::StringRef dContents;
+         section.getContents(dContents);
+         // If this library contains mangled name
+         return dContents.contains(mangled_name);
```

Comment:
When looking for a symbol `foo`, will this return "found!" if there is no `foo` symbol, only a `fooBar` symbol?